posts/2020-03-23-when-hibernate-caching-can-go-wrong.adoc

```diff
@@ -160,22 +160,22 @@ from department department0_
 left outer join company company1_ on department0_.company=company1_.id where department0_.id=?
 ```
 
-First time it queries employee, department and company as normal. 
+First time it queries employee, department, and company as normal.
 
-The second time it queries department and company tables.
+The second time it queries the department and company tables.
 
-So yes we cached Employee properly but we had cached only an id of department which an employee belongs to. Means, having this id, our application can either get an entity by id from an entity cache or it will go to the database again to gather missing data. Our department wasn't ever put to the entity cache so our app went to the DB.
+So yes, we cached Employee properly but we had cached only an id of the department which an employee belongs to. Having this id means our application can either get an entity by id from an entity cache or it will go to the database again to gather missing data. Our department wasn't ever put to the entity cache so our app went to the DB.
 
 === Takeaway:
 
-When you want to cache an entity, check all ...ToOne relations which are eagerly fetched by default. You either want to make them fetched lazily or you can also cache it's relation entities otherwise the queries to DB will be made to fetch missing data. Whatever works better for your project & data.
+When you want to cache an entity, check all ...ToOne relations which are eagerly fetched by default. You either want to make them fetched lazily or you can also cache its relation entities. Otherwise the queries to the DB will be made to fetch missing data. Whatever works better for your project and data.
 
 == Catch #3 (my favourite)
 
-=== query caching is killing the application performance
+=== Query caching is killing the application performance
 
-Let's change the set up for our entities, so they are not stored in the entity cache. Now we are going to use the query cache.
-To set up a query caching you need to explicitly add hints to enable query caching for each query and optionally specify the region where it is cached.
+Let's change the setup for our entities, so they are not stored in the entity cache. Now we are going to use the query cache.
+To set up query caching you need to explicitly add hints for each query and optionally specify the region where it's cached.
```

Comment:
Here I was specifying the hints: one for enabling query caching, the other one for specifying the region. So the sense is changed a bit now. I'd suggest the following:
``` To set up query caching you need to explicitly add hints for each query: one to enable query caching and another optional one to specify the region where it is cached.```

Reply:
Sounds good. Changed.